src/base/interfaces/istringable.h

```diff
@@ -28,13 +28,23 @@
 
 #pragma once
 
+#include <concepts>
+
 class QString;
 
+// This interface is provided for classes to help satisfy the `Stringable` concept,
```

Comment:
This sounds incorrectly. You confuse the purposes of interfaces (in terms of OOP) and concepts (that relate to templates etc.)

Reply:
Please suggest an alternative. I want the reader to clearly understand the relationship of this interface and the c++ concept below.

Reply:
They are, of course, related, but not in the way you present it. These are different (related to different programming paradigms) ways to achieve similar (but not always the same) goals.

Reply:
I understand but you didn't really help provide an alternative description. I'll wait and I hope you are not too obsessive over the language correctness that clouds your judgement over the documentation intention/purpose.

Reply:
But it is totally incorrect.
This interface does not help the class to satisfy the `Stringable` concept in any way. If you look at it from this angle, it even complicates the job, since it requires a little more code. To satisfy the Stringable concept, a class must implement `toString()` method, but inheriting `IStringable` interface does not help this in any way.

Reply:
You still didn't provide an description. Let us try another way: when the reader asks, what will you answer for these questions? I'll put them in the code comments.
1. Why do you provide both `IStringable` and `Stringable`?
2. When do you use `IStringable` (or `Stringable`)?

Reply:
FYI, my current review of related code showed that we are not using any advantages from `IStringable` interface right now. `loadValue()` can't use it for its intended purpose at all. `storeValue()` could, but then we need to provide a separate non-template overload for it.

Reply:
>FYI, my current review of related code showed that we are not using any advantages from IStringable interface right now.

Part of me consider `IStringable` is obsoleted by `Stringable`.
If we remove `IStringable` then we are back to this again: https://github.com/qbittorrent/qBittorrent/pull/19431#discussion_r1288603411
If there is some way to enforce/guarantee "concept is satisfied on some class" then I would embrace c++ concepts fully.

Reply:
> Let us try another way: when the reader asks, what will you answer for these questions? I'll put them in the code comments.
> 
> 1. Why do you provide both `IStringable` and `Stringable`?
> 2. When do you use `IStringable` (or `Stringable`)?

First of all, they should be declared in separate headers. I believe it is good enough way to separate things that belong to different programming paradigms.
As for answering the questions, first we need to decide what to do with the problem described in https://github.com/qbittorrent/qBittorrent/pull/19440#discussion_r1289862285.

Reply:
> Part of me consider `IStringable` is obsoleted by `Stringable`.

The advantage of `IStringable` would be that there are only one instance of `storeValue()` that serves all classes that implements it. Currently there is separate instance per class. Although, given its triviality, this shouldn't be a problem.

Reply:
> If there is some way to enforce/guarantee "concept is satisfied on some class" then I would embrace c++ concepts fully.

As I said before, it shouldn't be enforced/guaranteed generally. But if it is necessary in a particular use case, then adding an assertion is the appropriate way (provided that you put a comment next to it describing this particular use case).

Reply:
`IStringable` can be added back if needed sometime else.

Reply:
> In addition, it is not possible for a template class to add such assertion. You won't be able to cover all template parameters. And this poses a problem for `template <int N, int Mandatory = N> class Utils::Version`.

Anyway removing `IStringable` isn't source of this "problem". So if it is not being used (effectively) and there is no intention to use it in the near future, then we can safely delete it.

Reply:
PR updated. `IStringable` removed.